### PR TITLE
8262973: Verify ParCompactionManager instance in PCAdjustPointerClosure

### PR DESCRIPTION
**Pull request by albertnetymk**
Verify the correct ParCompactionManager instance is used by VM/GC threads and some general cleanup.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8262973](https://bugs.openjdk.java.net/browse/JDK-8262973): Verify ParCompactionManager instance in PCAdjustPointerClosure

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**) ⚠️ Review applies to 4a426d289be334731806b64985b9e553afe9ddf4

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2813/head:pull/2813`
`$ git checkout pull/2813`

### CODE REVIEW

File: src/hotspot/share/gc/parallel/psCompactionManager.cpp [hunk 1]
```diff
@@ -179,3 +179,19 @@ void ParCompactionManager::push_shadow_region(size_t shadow_region) {
 void ParCompactionManager::remove_all_shadow_regions() {
   _shadow_region_array->clear();
 }
+
+#ifdef ASSERT
+void ParCompactionManager::verify_all_marking_stack_empty() {
+  uint parallel_gc_threads = ParallelGCThreads;
+  for (uint i=0; i<=parallel_gc_threads; i++) {
```
Comment (kimbarrett) [inline src/hotspot/share/gc/parallel/psCompactionManager.cpp:186]: Use spaces around operators, here and line 193.

File: src/hotspot/share/gc/parallel/psParallelCompact.cpp [hunk 2]
```diff
@@ -2614,9 +2619,11 @@ void PSParallelCompact::compact() {
   }
 
   {
-    // Update the deferred objects, if any.  Any compaction manager can be used.
     GCTraceTime(Trace, gc, phases) tm("Deferred Updates", &_gc_timer);
-    ParCompactionManager* cm = ParCompactionManager::manager_array(0);
+    // Update the deferred objects, if any. Any compaction manager can be used.
+    // However, since the current thread is VM thread, we use the rightful one
+    // to keep the verification logic happy.
```
Comment (kimbarrett) [inline src/hotspot/share/gc/parallel/psParallelCompact.cpp:2625]: If using a different cm makes the verification logic unhappy, that suggests "Any compaction manager can be used" isn't exactly true.  Maybe something like "In principle, any compaction manager would do.  However, ..."

**Review by kimbarrett — APPROVED** [review]
A couple minor nits.  Otherwise looks good.

**Review by tschatzl — APPROVED** [review]
Lgtm apart from the issues Kim mentioned.

**Comment by albertnetymk** [comment]
All fixed as suggested.

**Review by kimbarrett — APPROVED** [review]

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**